### ISSUE TITLE
Incorrect ruamel import in __init__.py from v0.38.0

### ISSUE THREAD

**Issue by IL-William** [InfuseAI/piperider#923]
https://github.com/InfuseAI/piperider/blob/276a0c5138a1a715e20aea50dad86aead078f5e3/piperider_cli/yaml/__init__.py#L5

Hi there,
After installing `piperider` for the first time, I get the following ruaml issue

<img width="762" alt="Screenshot 2023-11-05 at 10 31 07" src="https://github.com/InfuseAI/piperider/assets/80535383/aaa1510f-1345-4f13-a6ae-c73e03673dec">

Looking at the specified file, I can see that the import is as follow

```python
from ruamel.yaml import StreamTextType, StreamType, VersionType
```

This file was inserted from `v0.38.0`.

I can see that it is specified that `ruaml` should be `< 0.17.40`.

However, I had a look at [ruamel repo using 0.17.40](https://sourceforge.net/p/ruamel-yaml/code/ci/0.17.40/tree/main.py), and I can see it was already using `from ruamel.yaml.compat`

<img width="767" alt="Screenshot 2023-11-05 at 10 33 38" src="https://github.com/InfuseAI/piperider/assets/80535383/fef21419-6c8f-447d-b25e-89807ffb29b1">

This means that the import should be as follows
```python
from ruamel.yaml.compat import StreamTextType, StreamType, VersionType
```

Alternatively, I don't know if the requested version should be downgraded to work as it is currently?

After manual modification, I can use `piperider`.

<img width="762" alt="Screenshot 2023-11-05 at 10 35 09" src="https://github.com/InfuseAI/piperider/assets/80535383/f355267d-1e9c-4543-8ab8-b84c2fe4430d">

In the meantime, I'll use `v0.37.1` as this file was not existing yet, and there is no issue currently with it.

**Comment by popcornylu** [comment]
Hi @IL-William, Thanks for your feedback. 

I have opened sc-32530 for internal tracking.

**Comment by popcornylu** [comment]
May be related to  #912

**Comment by popcornylu** [comment]
Fixed in release v0.40.0
